mllib/src/test/scala/org/apache/spark/ml/feature/CountVectorizerSuite.scala

```diff
@@ -183,6 +183,26 @@ class CountVectorizerSuite extends SparkFunSuite with MLlibTestSparkContext
       case Row(features: Vector, expected: Vector) =>
         assert(features ~== expected absTol 1e-14)
     }
+
+    // CountVectorizer test
+    val df2 = sqlContext.createDataFrame(Seq(
```

Comment:
@BryanCutler If I understand correctly, 

    val df = sqlContext.createDataFrame(Seq(
      (0, split("a a a b b c"), Vectors.sparse(4, Seq((0, 1.0), (1, 1.0), (2, 1.0)))),
      (1, split("c c c"), Vectors.sparse(4, Seq((2, 1.0)))),
      (2, split("a"), Vectors.sparse(4, Seq((0, 1.0))))
    )).toDF("id", "words", "expected")

    val cv = new CountVectorizerModel(Array("a", "b", "c", "d"))
      .setInputCol("words")
      .setOutputCol("features")
      .setBinary(true)
has different expectation if I use CountVectorizer to get the vocabulary, since the CountVectorizerModel(Array("a", "b", "c", "d")) takes a different dictionary.

So I can't reuse the DF. Am I right?

Thanks!

Miao